cdci_data_analysis/analysis/renku_helper.py

```diff
@@ -170,14 +173,57 @@ def create_new_notebook_with_code(repo, api_code, job_id, file_name=None):
     return file_path
 
 
-def commit_and_push_file(repo, file_path):
+def generate_commit_request_url(params_dic, products_url):
```

Comment:
can you reuse the function used elsewhere? I am sure this URL is constructed simularly in a different place.

Reply:
Yes it is, it's here: [https://github.com/oda-hub/dispatcher-app/blob/9ad3167723a116e4b485606feaf7c90dbc209075/cdci_data_analysis/flask_app/dispatcher_query.py#L1014](https://github.com/oda-hub/dispatcher-app/blob/9ad3167723a116e4b485606feaf7c90dbc209075/cdci_data_analysis/flask_app/dispatcher_query.py#L1014) .

I thought about it but this is an instance method of the `InstrumentQueryBackEnd`, sure we can make an instance of that class, but perhaps we add some overhead and complexity and it might become more complex to maintain.

What do you think?

Reply:
> Yes it is, it's here:
> 
> https://github.com/oda-hub/dispatcher-app/blob/9ad3167723a116e4b485606feaf7c90dbc209075/cdci_data_analysis/flask_app/dispatcher_query.py#L1014
> 
> This is an instance method of the `InstrumentQueryBackEnd` class. I thought about using it by making an instance of that class, but perhaps we add some overhead and complexity and it might become more complex to maintain.
> 
> What do you think?

I think `InstrumentQueryBackEnd.generate_products_url_from_par_dict` can be split in two parts:
* part which uses in self.use_scws - to remain in  InstrumentQueryBackEnd
* part which just generates url from par_dict  - to be taken out of the class in some other file.